### ISSUE TITLE
[BUG]: `llama2/tokenizer/ball.mojo` doesn't compile

### ISSUE THREAD

**Issue by soraros** [modularml/max#40]
### Bug description

Said file doesn't compile.
In order to run the file, an empty `main()` is added, and the compiler complains:
```
error: 'Optional' parameter #0 has 'CollectionElement' type, but value has type 'Node[T]'
```
If I were to import things defined in the module to other file, it runs fine.
```mojo
# main.mojo
from ball import Ball

fn main():
  ...
```

### Steps to reproduce

As above.

### System information

```shell
Mojo 24.1-beta on Docker, Intel Mac.
```

**Comment by goldiegadde** [comment]
@soraros thanks for reporting, this is now fixed, closing!!

**Comment by goldiegadde** [comment]
I accidentally closed this @soraros , sorry waiting for 1 more fix, reopening.

**Comment by goldiegadde** [comment]
@soraros this should be fixed with the 24.2 release. Closing this, please reopen if you continue to run into this.
